Update background tasks guide (#30814)
supabase/supabase: apps/docs/content/guides/functions/background-tasks.mdx

title: 'Background Tasks'
description: 'How to run background tasks in an Edge Function outside of the request handler'
subtitle: 'How to run background tasks in an Edge Function outside of the request handler'

> tocVideo: 'rSKBTJxG9VA'

Author: @thorwebdev I removed the video from the guide as the syntax has changed (no longer need to do custom events). Would you be able to do an updated version of the video?

Edge Function instances can process background tasks outside of the request handler. Background tasks are useful for asynchronous operations like uploading a file to Storage, updating a database, or sending events to a logging service. You can respond to the request immediately and leave the task running in the background.

### How it works

You can use `EdgeRuntime.waitUntil(promise)` to explicitly mark background tasks. Function instance would continue to run until the promise provided to `waitUntil` completes.

> The maximum duration it will run will be based on wall-clock, CPU or memory limit. The Function will shutdown when it reaches one of the [limits](/docs/guides/functions/limits).

Reviewer: ```suggestion
The maximum duration is capped based on the wall-clock, CPU, and memory limits. The Function will shutdown when it reaches one of these [limits](/docs/guides/functions/limits).
```

> You can listen to `beforeunload` event handler to be notified when function invocation is about to be shutdown.

Reviewer: ```suggestion
You can listen to the `beforeunload` event handler to be notified when Function invocation is about to be shut down.
```

### Example

> Here's an example of using `EdgeRuntime.waitUntil` to run a background task and using `beforeunload` event to be notified when the instance is about to be shutdown.

Reviewer: ```suggestion
Here's an example of using `EdgeRuntime.waitUntil` to run a background task and using `beforeunload` event to be notified when the instance is about to be shut down.
```
